### ISSUE TITLE
Cannot commit parts of the code to other branch because it's "locked"

### ISSUE THREAD

**Issue by stijnvanderlaan** [gitbutlerapp/gitbutler#3881]
I can't commit bulletinBoardItems (new line) part of the code to another branch, because it's "locked".

<img width="455" alt="Screenshot 2024-05-28 at 14 56 08" src="https://github.com/gitbutlerapp/gitbutler/assets/20322260/17aa4644-e997-45aa-b8d2-109f9c427e15">

**Comment by Byron** [comment]
Could you also show a screenshot of the vbranches, with the respective file selected? 

Locked hunks can be legitimate if the hunk is already committed to another vbranch. I'd like to find out if in your case, the hunk should not be locked.

**Comment by stijnvanderlaan** [comment]
Isn't it true that a new line should never be "legitimate" locked, in any case? It's not related to committed anywhere. I think it had something to do with unrelated code which IS related to the branch it was locked on. In the screenshot below: archiveYear etc is a change already committed to another branch but the bulletingBoardItems is unrelated and should be committed to another branch, undo the changes to archiveYear etc results in the scenario described in this ticket.

<img width="443" alt="Screenshot 2024-05-28 at 14 50 02" src="https://github.com/gitbutlerapp/gitbutler/assets/20322260/4ebb981a-1e2d-4aba-a245-a2412cf808a2">

**Comment by Byron** [comment]
Thank you! I imagined a different screenshot to make sense of it, but admittedly I am also not the fittest in how hunk locking is working or is supposed to work.
Maybe @krlvi can understand better what's going on here - maybe it's something about the *newlines* as well?

**Comment by mtsgrd** [comment]
@stijnvanderlaan you're right in that they are unrelated changes, but unfortunately we don't yet support splitting hunks across branches. Under the hood we work with patches in order to create of a unified integration commit, as well as the individual branches that get pushed to the remote. So to maintain compatibility between branches you currently need changes for different branches to be separated by enough unchanged lines (7 to be precise) so it shows up as two separate diffs in the UI.

But stay tuned, it should be possible for us to support line-by-line splitting in the future.

**Comment by krlvi** [comment]
Hey @stijnvanderlaan, this is supported now. https://github.com/gitbutlerapp/gitbutler/pull/8068 added partial committing, and https://github.com/gitbutlerapp/gitbutler/pull/8205 made dependency locks line-level, so unrelated lines can be selected and committed to the branch you want without taking neighboring locked changes with them.

Both shipped by `0.14.19`: https://github.com/gitbutlerapp/gitbutler/releases/tag/release/0.14.19

Sorry this sat open for so long.
